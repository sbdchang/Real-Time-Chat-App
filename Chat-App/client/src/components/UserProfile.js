import React from 'react';
import PageNavbar from './PageNavbar';
import 'bootstrap/dist/css/bootstrap.min.css';
import avatar from './avatar.png';
const axios = require('axios');

export default class UserProfile extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			username: "hzw",
			date: "Registration Time"
		};
	}

	componentDidMount() {
		// this.setState({
		// 	username: this.props.changedUsername
		// });
		fetch(`http://localhost:8081/users?username=${this.state.username}`, {
			method: "GET" 
		}).then(response => response.json()).then((response) => {
			this.setState({
				date: response.dateRegistered.substr(0, response.dateRegistered.indexOf('T'))
			});
		})
	}

	async changePassword() {
		let messageOne = document.querySelector("#message-1");
		messageOne.textContent = "";
		await fetch(`http://localhost:8081/users/cpw`, {
			method: "POST" 
		}).then((response) => {
			if (document.getElementById('cpw').value === document.getElementById('npw').value) {
				messageOne.textContent = "Password NOT Changed";
			} else {
				messageOne.textContent = "Password Changed";
			}
		})
	}

	render() {	
		return (
			<div className="UserProfile">
				<PageNavbar active="userprofile" />
				<div className="container bestgenres-container">
			      <div className="jumbotron">
		            <div className="h5">{`Username: ${this.state.username}`}</div>
					<div className="h6">{`Regestration Time: ${this.state.date}`}</div>
					<img src={avatar}/>
			      </div>

			      <div className="jumbotron">
				  <input type='text' placeholder="Current Password" id="cpw" className="cpw-input"/>
				  <input type='text' placeholder="New Password" id="npw" className="npw-input"/>
				  <button id="cp" className="cp-btn" onClick={this.changePassword}>Change Password</button>
				  <div className="results-container" id="results">
			    		<p id = "message-1">  </p>
			      </div>
				  <br></br>
				  <input type='text' placeholder="Current Password" id="dcpw" className="cpw-input"/>
				  <button id="da" className="da-btn">Deactivate Account</button>
				  <div className="results-container" id="results">
			    		<p id = "message-2">  </p>
			      </div>
			      </div>
			    </div>
			</div>
		);
	}
}