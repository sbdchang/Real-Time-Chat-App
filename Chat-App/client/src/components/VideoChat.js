import React, { useState, useCallback, useEffect } from 'react';
import {urlToUse} from "./url";
import Lobby from './Lobby';
import Room from './Room';

const VideoChat = () => {
  const [username, setUsername] = useState('');
  const [roomName, setRoomName] = useState('');
  const [token, setToken] = useState(null);



  // useEffect(() => {
  //   const queryString = window.location.search;
  //   const urlParams = new URLSearchParams(queryString);
  //   const room_name = urlParams.get('room_name');
  //   const user_name = urlParams.get('user_name');
  //   setUsername(user_name);
  //   setRoomName(room_name);
  //   const data = fetch(`${urlToUse.url.API_URL}/video/token`, {
  //     method: 'POST',
  //     body: JSON.stringify({
  //       identity: username,
  //       room: roomName
  //     }),
  //     headers: {
  //       'Content-Type': 'application/json'
  //     }
  //   }).then(res => {
  //     const data = res.json();
  //     console.log(data.token);
  //     setToken(data.token);
  //   });
  // }, []);

  const handleUsernameChange = useCallback(event => {
    setUsername(event);
  }, []);

  const handleRoomNameChange = useCallback(event => {
    setRoomName(event);
  }, []);

  const handleSubmit = useCallback(
    async () => {
      // event.preventDefault();
      const data = await fetch(`${urlToUse.url.API_URL}/video/token`, {
        method: 'POST',
        body: JSON.stringify({
          identity: username + "asdfvew",
          room: roomName
        }),
        headers: {
          'Content-Type': 'application/json'
        }
      }).then(res => res.json());
      setToken(data.token);
    },
    [roomName, username]
  );

  const handleLogout = useCallback(event => {
    window.location.replace(`${document.location.origin}/mainview?username=${username}`);
  }, [username]);

  let render;
  if (token) {
    render = (
      <Room roomName={roomName} token={token} handleLogout={handleLogout} />
    );
  } else {
    render = (
      <Lobby
        username={username}
        roomName={roomName}
        handleUsernameChange={handleUsernameChange}
        handleRoomNameChange={handleRoomNameChange}
        handleSubmit={handleSubmit}
      />
    );
  }
  return render;
};

export default VideoChat;
