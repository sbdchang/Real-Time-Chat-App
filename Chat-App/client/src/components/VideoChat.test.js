import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import UserProfile from './UserProfile';
import renderer from 'react-test-renderer';
import VideoChat from './VideoChat';
import MainView from './MainView';

it('Videochat Snapshot  ', () => {
    const tree = renderer.create(<MainView />).toJSON();
    expect(tree).toMatchSnapshot();
  });


// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<MainView />, div);
// });
