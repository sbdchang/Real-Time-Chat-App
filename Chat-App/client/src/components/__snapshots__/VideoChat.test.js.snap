// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videochat Snapshot   1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Enter a room
  </h2>
  <div>
    <label
      htmlFor="name"
    >
      Name:
    </label>
    <input
      id="field"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
  </div>
  <div>
    <label
      htmlFor="room"
    >
      Room name:
    </label>
    <input
      id="room"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
  </div>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
